File: 3754/solution.py
class Solution:
    def sumAndMultiply(self, n: int) -> int:
        x, tot, i = 0, 0, 1 
        while n:
            d = n%10 
            n//=10 
            if not d:
                continue
            x = d*i + x
            tot+=d 
            i*=10
        return x*tot

File: 3754/solution.java
class Solution {
    public long sumAndMultiply(int n) {
        int x = 0;
        int sum = 0;
        int i = 1;
        int d;
        while(n>0){
            d = n%10;
            n/=10;
            if(d==0)
                continue;
            x = d*i + x;
            sum+=d;
            i*=10;
        }
        return (long)sum*x;
    }
}
